sklearn/preprocessing/data.py

```diff
@@ -609,13 +610,19 @@ def normalize(X, norm='l2', axis=1, copy=True):
             inplace_csr_row_normalize_l1(X)
         elif norm == 'l2':
             inplace_csr_row_normalize_l2(X)
+        elif norm == 'max':
+            _, norms = min_max_axis(X, 1)
+            norms = norms.repeat(np.diff(X.indptr))
+            mask = norms != 0
+            X.data[mask] /= norms[mask]
```

Comment:
I feel like we should raise a ValueError "else". If you could add that and add a test, it would be much appreciated.